### PR TITLE
refactor(pkg): remove timeout in WaitTillL2ExecutionEngineSynced

### PR DESCRIPTION
**Pull request by YoGhurt111**

### CODE REVIEW

File: pkg/rpc/methods.go [hunk 1]
```diff
@@ -373,13 +370,9 @@ func (c *Client) GetProtocolStateVariables(opts *bind.CallOpts) (*struct {
 		opts = &bind.CallOpts{}
 	}
 
-	var ctx = context.Background()
-	if opts.Context != nil {
-		ctx = opts.Context
+	if opts.Context == nil {
+		opts.Context = context.Background()
 	}
-	ctxWithTimeout, cancel := context.WithTimeout(ctx, defaultWaitReceiptTimeout)
```
Comment (davidtaikocha) [inline pkg/rpc/methods.go:380]: Why we need to remove the timeout config here?

File: pkg/rpc/methods.go [hunk 2]
```diff
@@ -101,6 +104,7 @@ func (c *Client) WaitTillL2ExecutionEngineSynced(ctx context.Context) error {
 					"currentBlockID", progress.CurrentBlockID,
 					"highestBlockID", progress.HighestBlockID,
 					"progress", progress.SyncProgress,
+					"Time-consuming statistics", time.Since(start),
```
Comment (davidtaikocha) [inline pkg/rpc/methods.go:107]:
```suggestion
					"time", time.Since(start),
```

File: pkg/rpc/methods.go [hunk 3]
```diff
@@ -81,15 +81,18 @@ func (c *Client) ensureGenesisMatched(ctx context.Context) error {
 
 // WaitTillL2ExecutionEngineSynced keeps waiting until the L2 execution engine is fully synced.
 func (c *Client) WaitTillL2ExecutionEngineSynced(ctx context.Context) error {
-	if ctx.Err() != nil {
+	if ctx.Err() != nil && !errors.Is(ctx.Err(), context.DeadlineExceeded) {
```
Comment (davidtaikocha) [inline pkg/rpc/methods.go:84]: Lets add a test for this?

File: pkg/rpc/methods.go [hunk 4]
```diff
@@ -81,15 +81,18 @@ func (c *Client) ensureGenesisMatched(ctx context.Context) error {
...
 		return ctx.Err()
 	}
+	newCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	start := time.Now()
 	return backoff.Retry(
 		func() error {
-			if ctx.Err() != nil {
+			if ctx.Err() != nil && !errors.Is(ctx.Err(), context.DeadlineExceeded) {
```
Comment (davidtaikocha) [inline pkg/rpc/methods.go:92]:
```suggestion
			if ctx.Err() != nil && !errors.Is(ctx.Err(), context.Background()) {
```

Seems this should also be ok?

File: pkg/rpc/methods.go [hunk 5]
```diff
@@ -81,15 +81,18 @@ func (c *Client) ensureGenesisMatched(ctx context.Context) error {
...
 				return ctx.Err()
 			}
-			progress, err := c.L2ExecutionEngineSyncProgress(ctx)
+			progress, err := c.L2ExecutionEngineSyncProgress(newCtx)
```
Comment (davidtaikocha) [inline pkg/rpc/methods.go:95]:
```suggestion
			progress, err := c.L2ExecutionEngineSyncProgress(context.Background())
```

Comment (davidtaikocha) [reply pkg/rpc/methods.go:95]: Seems this should also be ok?

File: pkg/rpc/methods.go (hunk 3 above)
Comment (YoGhurt111) [reply pkg/rpc/methods.go:84]: OK

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**